src/hidimstat/desparsified_lasso.py

```diff
@@ -152,271 +119,374 @@ def desparsified_lasso(
-        X, Y, method=noise_method, order=order, n_jobs=n_jobs
-    )
+def _compute_residuals(X, column_index, alpha, gram, max_iter=5000, tol=1e-3):
+    """
+    Compute nodewise Lasso regression for desparsified Lasso estimation
 
-    if cov is not None:
-        cov_hat = cov
+    For feature i, regresses X[:,i] against all other features to
+    obtain residuals and precision matrix diagonal entry needed for debiasing.
 
-    theta_hat = n_samples * inv(cov_hat)
+    Parameters
+    ----------
+    X : ndarray, shape (n_samples, n_features)
+        Centered input data matrix
 
-    # Estimating the coefficient vector
-    beta_bias = Y.T.dot(Z) / np.sum(X * Z, axis=0)
+    column_index : int
+        Index i of feature to regress
 
-    beta_mtl = beta_mtl.T
-    beta_bias = beta_bias.T
+    alpha : float
+        Lasso regularization parameter
 
-    P = (np.dot(X.T, Z) / np.sum(X * Z, axis=0)).T
-    P_nodiag = P - np.diag(np.diag(P))
+    gram : ndarray, shape (n_features, n_features)
+        Precomputed X.T @ X matrix
 
-    beta_hat = beta_bias - P_nodiag.dot(beta_mtl)
+    max_iter : int, default=5000
+        Maximum Lasso iterations
 
-    if test == "chi2":
+    tol : float, default=1e-3
+        Optimization tolerance
 
-        chi2_scores = np.diag(multi_dot([beta_hat, theta_hat, beta_hat.T])) / omega_diag
-        two_sided_pval = np.minimum(2 * stats.chi2.sf(chi2_scores, df=n_times), 1.0)
+    Returns
+    -------
+    z : ndarray, shape (n_samples,)
+        Residuals from regression
 
-    if test == "F":
+    omega_diag_i : float
+        Diagonal entry i of precision matrix estimate,
+        computed as n * ||z||^2 / <x_i, z>^2
 
-        f_scores = (
-            np.diag(multi_dot([beta_hat, theta_hat, beta_hat.T])) / omega_diag / n_times
-        )
-        two_sided_pval = np.minimum(
-            2 * stats.f.sf(f_scores, dfd=n_samples, dfn=n_times), 1.0
-        )
+    Notes
+    -----
+    Uses sklearn's Lasso with precomputed Gram matrix for efficiency.
+    """
 
-    sign_beta = np.sign(np.sum(beta_hat, axis=1))
-    pval, pval_corr, one_minus_pval, one_minus_pval_corr = (
-        pval_from_two_sided_pval_and_sign(two_sided_pval, sign_beta)
-    )
+    n_samples, n_features = X.shape
+    i = column_index
 
-    return beta_hat, pval, pval_corr, one_minus_pval, one_minus_pval_corr
+    # Removing the column to regress against the others
+    X_new = np.delete(X, i, axis=1)
+    y_new = np.copy(X[:, i])
+
+    # Method used for computing the residuals of the Nodewise Lasso.
+    # here we use the Lasso method
+    gram_ = np.delete(np.delete(gram, i, axis=0), i, axis=1)
+    clf = Lasso(alpha=alpha, precompute=gram_, max_iter=max_iter, tol=tol)
+
+    # Fitting the Lasso model and computing the residuals
+    clf.fit(X_new, y_new)
+    z = y_new - clf.predict(X_new)
+
+    # Computing the diagonal of the covariance matrix
+    omega_diag_i = n_samples * np.sum(z**2) / np.dot(y_new, z) ** 2
```

Comment:
Maybe it would help to clarify that this refers to the noise covariance ?

Reply:
why not in the comment just above?

Reply:
Sorry, I badly express myself. I updated the comment and not the docstring.

Reply:
@jpaillard